### PR TITLE
Convert the ResourceQuotaStatus proposal into the proper documentation

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Conversion of the ResourceQuotaStatus proposal to documentation.

### CODE REVIEW

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# ResourceQuotasStatus 
+
+The ResourceQuotasStatus contains the flag and list of exceeded ResourceQuotas limits, together with a set of resources which exceed that limits. The value of the flag informs you if any ResourceQuota is exceeded in any possible way.
+
+ResourceQuotasStatus should detect whenever any ReplicaSet or StatefulSet in your Environment cannot progress because it's blocked by the ResourceQuota.
+To check that you must calculate the required amount of resources to create another replica. If there is not enough resources, you should return the exceeded status.
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js#L86) you can find the example implementation of the checking logic in UI.
```
Comment (aszecowka) [inline components/ui-api-layer/docs/resource-quotas-status.md]: I think that this link is not necessary

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+The ResourceQuota values can be as follows:
+```bash
+Name:            kyma-default
+Resource         Used       Hard
+--------         ---        ---
+limits.memory    7348440Ki  10Gi
+requests.memory  3727576Ki  7Gi
+```
+The `used` values of the ResourceQuota can exceed the `hard` values and block any resource creation in the Environment.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: ``When the `used` values of the ResourceQuota exceed the `hard` values, the resource creation in the Environment is blocked.``

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
+# ResourceQuotasStatus 
+
+[ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/) provides constraints that limit aggregate resource consumption per Namespace.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `the aggregate resource consumption`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+```bash
+Name:            kyma-default
+Resource         Used       Hard
+--------         ---        ---
+limits.memory    7348440Ki  10Gi
+requests.memory  3727576Ki  7Gi
+```
+The `used` values of the ResourceQuota can exceed the `hard` values and block any resource creation in the Environment.
+
+The ResourceQuotasStatus contains the flag and list of exceeded ResourceQuotas limits, together with a set of resources which exceed that limits. The value of the flag informs you if any ResourceQuota is exceeded in any possible way.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `the list of`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+ResourceQuotasStatus should detect whenever any ReplicaSet or StatefulSet in your Environment cannot progress because it's blocked by the ResourceQuota.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `The ResourceQuotasStatus detects if any ReplicaSet or StatefulSet in your Environment is blocked by the ResourceQuota and cannot progress.`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+To check that you must calculate the required amount of resources to create another replica. If there is not enough resources, you should return the exceeded status.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: ``To check if any ReplicaSet or StatefulSet is blocked, calculate the required number of resources to create another replica. If there are not enough resources to create another replica, return the ResourceQuotaStatus with the flag set to `true`.``

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js#L86) you can find the example implementation of the checking logic in UI.
+
+## The Console calls
+
+The Console calls for the ResourceQuotasStatus automatically. The calls are triggered after: 
+- switching the environment
+- uploading a resource
+- creating a lambda 
+- opening environment's details
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `Environment's`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+## Implementation
+
+This section contains the steps of the ResourceQuotaStatus implementation.
+
+### Calculate the available resources in the Environment
+
+To calculate amount of available resources in the given Environment you must list the ResourceQuotas and loop through them.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `To calculate the number of available resources in the given Environment,  list the ResourceQuotas and loop through them.`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+You can get the available amount of resources by calculating the difference from `.spec.hard` and `.status.used` in each ResourceQuota.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `the available number of`
`**.spec.hard** and **.status.used**`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+For each resource limit specified in the ResourceQuotas, you must calculate the available amount of resources using the ResourceQuota with the lowest `.spec.hard` value for that resource type.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `the available number of`
`**.spec.hard**`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+### Calculate the necessary resources to create the next replica
+
+To calculate how much resources you need to create the next replica, you must sum up the resource usage of all containers in the replica Pod.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `To calculate how much resources you need to create the next replica, sum up the resource usage of all containers in the replica Pod.`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+If some resource limit is not specified directly in the replica's template, you must check if any LimitRange is not specifying it.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: ~you must~ `check if any LimitRange does not specify it.`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+In this situation you must add the proper LimitRange limit to the list of necessary resources.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `In this situation, add the proper LimitRange limit to the list of necessary resources.`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+When the same limit is specified in the LimitRange and in the replica template, the replica's template limit has a priority.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: `has priority.`

File: components/ui-api-layer/docs/resource-quotas-status.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+### Check the StatefulSets
+
+List StatefulSets in the given Namespace and loop through them. You must check every StatefulSet which does not reach the number of desired replicas.
+To achieve that, implement the logic which calculates if the StatefulSet has enough resources to create the next replica.
+If there is not enough resources to create the next replica and the number of replicas is not reached, return a ResourceQuotasStatus with a flag set to `true`.
+
+### End of checking
+
+When both checks passed and the resources usage in the given Environment did not exceed any ResourceQuota limit, return the ResourceQuotasStatus with a flag set to `false`.
```
Comment (klaudiagrz) [inline components/ui-api-layer/docs/resource-quotas-status.md]: ``When both checks have passed and the resources usage in the given Environment did not exceed any ResourceQuota limit, return the ResourceQuotasStatus with a flag set to `false`.``

**Comment by polskikiel** [comment]
@klaudiagrz Fixed that `DOC` after your review. Please, merge this PR if it suits you.

**Pull request merged**